### PR TITLE
Optimization for code

### PR DESCRIPTION
**Pull request by bikevit2008**
This code works good, but without unnecessary else if

## Description

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../CONTRIBUTIONS.md).

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

File: packages/web3-utils/src/utils.js [hunk 1]
```diff
@@ -84,13 +84,11 @@ var toTwosComplement = function (number) {
  */
 var isAddress = function (address) {
     // check if it has the basic requirements of an address
-    if (!/^(0x)?[0-9a-f]{40}$/i.test(address)) {
-        return false;
-        // If it's ALL lowercase or ALL upppercase
-    } else if (/^(0x|0X)?[0-9a-f]{40}$/.test(address) || /^(0x|0X)?[0-9A-F]{40}$/.test(address)) {
+    //case insensitive regexp
+    if (/^(0x)?[0-9a-f]{40}$/i.test(address)) {
```
Comment (nazarhussain) [inline packages/web3-utils/src/utils.js:88]: The earlier logic was only accepting either all upper case or all lower case. This change will allow mixed case addresses. I don't think it's appropriate to change.

**Review by nazarhussain — CHANGES_REQUESTED** [review]

File: packages/web3-utils/src/utils.js (hunk 1 above)
Comment (jdevcs) [reply packages/web3-utils/src/utils.js:88]: @nazarhussain checksum addresses are mixed case so its valid

**Comment by jdevcs** [comment]
@bikevit2008  could you update CHANGELOG.md, under 1.7.2

**Review by jdevcs — CHANGES_REQUESTED** [review]
Thanks @bikevit2008 . could you check there are tests failing with following messages :
```
TypeError: address.replace is not a function
Cannot read property 'toLowerCase' of undefined
.....
```
and also update CHANGELOG.md, and mention this PR under 1.7.2

**Review by nazarhussain — APPROVED** [review]

**Comment by nazarhussain** [comment]
@bikevit2008 Please check the failing tests and also update teh CHANGELOG.md as suggested by @jdevcs

**Comment by jdevcs** [comment]
> Thanks @bikevit2008 . could you check there are tests failing with following messages :
> 
> ```
> TypeError: address.replace is not a function
> Cannot read property 'toLowerCase' of undefined
> .....
> ```
> 
> and also update CHANGELOG.md, and mention this PR under 1.7.2

Following are failing and should to be fixed, 
Build / unit (14) (pull_request) 
Build / e2e (unit_and_e2e_clients) (pull_request) 
Build / e2e (e2e_browsers) (pull_request)
Build / e2e (e2e_ganache) (pull_request) 
Build / e2e_windows (pull_request)

There is no activity by author on this PR so closing,  @bikevit2008 feel free to reopen this PR when you have time for contributions.

**Pull request closed without merging**